module/preference/PreferenceDialog.kt

```diff
@@ -346,16 +350,23 @@ object PreferenceDialog {
                 caption = PREFERENCES_GRAPH_OUTPUT_CAPTION
             )
 
-            override val preferences: List<PreferenceField> = listOf(graphOutput)
+            private val graphThemes = listOf(Color.Themes.DARK_GRAPH, Color.Themes.LIGHT_GRAPH)
+            private var graphTheme = PreferenceField.Dropdown(
+                initValue = preferenceMgr.graphTheme, values = graphThemes, label = SET_GRAPH_THEME,
+            )
+
+            override val preferences: List<PreferenceField> = listOf(graphOutput, graphTheme)
 
             override fun submit() {
                 preferenceMgr.graphOutputEnabled = graphOutput.value
-                graphOutput.modified = false
+                preferenceMgr.graphTheme = graphTheme.value
+                preferences.forEach { it.modified = false }
```

Comment:
This line looks like it's going to be repeated a _lot_ as the preference dialog grows.

We could move it up to the base function `submit` (and `reset`) but that doesn't really improve the situation much: we now need to call `super.submit()` in each `submit` method, and we need to trust that it's called in the right order and that future `PreferenceGroup` subclasses don't forget to write that line. Ouch.

Maybe the root of the problem is a suboptimal `Preference` model.

Each `submit` button largely does the same thing: commit all preference view states to preference manager, then mark all preferences as unmodified. The same goes for `reset`.

Could we refactor `PreferenceGroup` such that we no longer need to override `submit` and `reset` at all, because they simply iterate through all preferences declared in some kind of list, and commit or reset them as appropriate, then mark them all as unmodified?